pom.xml

```diff
@@ -471,6 +477,22 @@
         </pluginManagement>
 
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>analyze</id>
+                        <goals>
+                            <goal>analyze-only</goal>
+                        </goals>
+                        <configuration>
+                            <failOnWarning>true</failOnWarning>
+                            <ignoreNonCompile>true</ignoreNonCompile>
```

Comment:
I've relaxed warnings here for dependencies not in compile scope - this means that we might pull in unused dependencies at runtime/provided/test scope but that seems like a good compromise.